### ISSUE TITLE
Vim motion "dt}" (delete till closing bracket) deletes too much.

### ISSUE THREAD

**Issue by fasterthanlime** [zed-industries/zed#13788]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In the following code:

```toml
[dependencies]
pin-project-lite = { version = "0.2.14", optional = true }
                                     # 👆 cursor is here
clap = { version = "4.5.8", optional = true, features = ["derive"] }

[dev-dependencies]
```

...I would expect "dt}" to result in this diff:

```diff
diff --git a/Cargo.lock b/Cargo.lock
index 25dd2b3..9eba30e 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -632,6 +632,7 @@ dependencies = [
  "clap",
  "eyre",
  "libloading",
+ "pin-project-lite",
  "serde",
  "tokio",
  "tracing",
diff --git a/conflux/Cargo.toml b/conflux/Cargo.toml
index 7d64449..b9d4602 100644
--- a/conflux/Cargo.toml
+++ b/conflux/Cargo.toml
@@ -11,7 +11,7 @@ serde = { version = "1.0.203", optional = true, features = ["derive"] }
 libloading = { version = "0.8.4", optional = true }
 tracing = { version = "0.1.40" }
 tokio = { version = "1.38.0", features = ["full"] }
-pin-project-lite = { version = "0.2.14", optional = true }
+pin-project-lite = { version = "0.2.14" }
 clap = { version = "4.5.8", optional = true, features = ["derive"] }
 
 [dev-dependencies]
```

But instead it results in this diff:

```diff
diff --git a/conflux/Cargo.toml b/conflux/Cargo.toml
index 7d64449..3cbf40f 100644
--- a/conflux/Cargo.toml
+++ b/conflux/Cargo.toml
@@ -11,9 +11,6 @@ serde = { version = "1.0.203", optional = true, features = ["derive"] }
 libloading = { version = "0.8.4", optional = true }
 tracing = { version = "0.1.40" }
 tokio = { version = "1.38.0", features = ["full"] }
-pin-project-lite = { version = "0.2.14", optional = true }
-clap = { version = "4.5.8", optional = true, features = ["derive"] }
-
 [dev-dependencies]
 serde_json = "1.0.118"
```

Note that **this only seems to happen if I'm fast**.

If I mark a pause between "dt" and "}" (which is "Maj ]" on my US macbook pro keyboard), it works as expected! See for yourself:

https://github.com/zed-industries/zed/assets/7998310/8d390fc3-2394-4cf1-9797-3c245e539102

### Environment

Zed: v0.142.4 (Zed Preview)
OS: macOS 14.5.0
Memory: 8 GiB
Architecture: aarch64

**Comment by mrnugget** [comment]
I cannot reproduce this, neither in Preview nor in Stable.

https://github.com/zed-industries/zed/assets/1185253/9717d467-2e2f-494b-9fa1-ff9a5100d901

Same question as in the other ticket:

1. Does it reproduce with an empty `keymap.json`?
2. What keyboard layout do you have?

**Comment by fasterthanlime** [comment]
I do have vim mode normal mode bindings!! See my current key map: https://github.com/fasterthanlime/dotfiles/blob/main/.config/zed/keymap.json (I would try to reproduce but I’m away from my computer right now)

I also didn’t even think about that but my keyboard layout is Polish (QWERTY, so I see no difference from US most of the time)

**Comment by fasterthanlime** [comment]
This is related to my keymap, much like #13789, see:

 https://github.com/zed-industries/zed/issues/13789#issuecomment-2213450190

**Comment by mrnugget** [comment]
This should be fixed by https://github.com/zed-industries/zed/pull/14233. Closing, but happy to reopen if it didn't. But pretty sure it did.

**Comment by fasterthanlime** [comment]
Just built zed from source to confirm — this does appear fixed (as well as #13789).

Thanks for letting me keep my weird keymap!

**Comment by mrnugget** [comment]
> Thanks for letting me keep my weird keymap!

Hey, no judgement over here 😎
